### PR TITLE
Feature | Copy Link to Activity

### PR DESCRIPTION
**Pull request by davidebukali**
Implement link to activity on [report page](https://trello.com/c/ho7kIpkS/294-implement-link-to-activity-query-parameter-on-report-page)

### CODE REVIEW

**Review by edwinmp — CHANGES_REQUESTED** [review]
Good work on this. Just some little feedback.

image

- I can't get the copy feature to work ... there's an error in my console that `copyText is not a function`.
- The copy button should always be showing, otherwise users won't know  it's there. I have a feeling we may go for a small button that reads `Copy link`. Keep as is for now and we'll see after.
- Replace the tooltip with simple text i.e. when a user clicks the copy button, replace the button with a `Copied`. This should be ephemeral i.e. it returns to normal after 2 seconds. Current one stays forever. You can ditch popper. Keep the tick icon.
- Should be able to filter by name and by ID. So just have the placeholder as `Search ...` and implement the filter by ID (names are not unique so search by name only doesn't add much value).

image

**Review by edwinmp — CHANGES_REQUESTED** [review]
1. All the copy button should do is `copy` the link. Nothing else. So clicking the copy button should NOT update the URL and force a filter.
2. When a user enters `http://localhost:4200/report/jica-tc_2012?id=XM-DAC-701-8-1977000539` in their browser address bar & presses enter, the following should happen:

- The search bar should be populated with the id taken from the URL
- The report should be filtered to only show the activity with that matching ID, if any

**Review by nosvalds — CHANGES_REQUESTED** [review]
Nice job!

## Review

- The main feature works as expected, navigating to `http://localhost:4200/report/jica-088_2014?id=XM-DAC-701-8-2014959957` filters so that we only see the activity with id `XM-DAC-701-8-2014959957`
- The case where the file has some activities with errors, but the user tries to navigate to an activity with no errors (or to an activity that doesn't exist in the file) works as expected. You get a `No matching activities found` message

image

- The case where the file has no errors in any activities also works as well, `http://localhost:4200/report/fifty_eight-et?id=ABC-123` because in this case the filter doesn't even show up, so nothing happens with the query parameter as expected.

## Issues/Improvements

- The "Search and filter box" needs to give some indication of what it's searching and filtering on for the user. It does not seem to search for error messages, which is okay, but could be confusing because the filters can filter by error message/category, etc. This could be accomplished in the search box. So instead of `Search...` have it be `Search activity title and identifier...`. 

Or if there is not enough room for `Search activity title and identifier...` then maybe have a `?` icon with a tooltip to either side of the search box that says `searches activity titles and identifiers`

image

File: src/components/report/DocumentReport.vue [hunk 1]
```diff
@@ -108,15 +127,46 @@
...
 <template>
   <div class="-mx-3.5 flex flex-wrap">
     <div v-if="hasMessages" class="relative flex shrink grow flex-col sm:w-full md:basis-1/3">
       <div class="sticky top-0 m-2.5">
-        <h3 class="text-xl font-bold">Filters</h3>
+        <h3 class="text-xl font-bold">Search and filter</h3>
         <div class="bg-slate-300">
           <div class="px-4 py-2">
+            <SearchFilter
+              placeholder="Search ..."
```
Comment (nosvalds) [inline src/components/report/DocumentReport.vue:163]:
```suggestion
              placeholder="Search title and identifier..."
```

@davidebukali can you make this change and then I'll approve the PR? I tried but I don't have access rights on the devinit fork.

Comment (nosvalds) [reply src/components/report/DocumentReport.vue]: Looks good.

**Pull request merged**